components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONRequestParser.java

```diff
@@ -105,113 +104,110 @@ public static RequestCtx parse(String jsonRequest) throws JsonParseException, Re
-                                    .getAsJsonObject()));
-                        }
+    private static void jsonAttributeSeperator(Map.Entry<String, JsonElement> jsonAttribute, JsonObject jsonCategory, Set<Attributes> categories, MultiRequests multiRequests) throws RequestParseException, UnknownIdentifierException {
+
+        Node content = null;
+        URI category = null;
+        Set<Attribute> attributes = null;
+        String id = null;
+        switch (jsonAttribute.getKey()) {
+
+            // For a custom Category
+            // Or a Category with long identifier
+            case EntitlementEndpointConstants.CATEGORY_DEFAULT:
+                if (jsonCategory.has(EntitlementEndpointConstants.CATEGORY_ID)) {
+                    category = stringCateogryToURI(jsonCategory
+                            .get(EntitlementEndpointConstants.CATEGORY_ID)
+                            .getAsString());
+                }
+            case EntitlementEndpointConstants.CATEGORY_RESOURCE:
+            case EntitlementEndpointConstants.CATEGORY_ACTION:
+            case EntitlementEndpointConstants.CATEGORY_ENVIRONMENT:
+            case EntitlementEndpointConstants.CATEGORY_ACCESS_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_RECIPIENT_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_INTERMEDIARY_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_CODEBASE:
+            case EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE:
 
-                        //multiRequests = new MultiRequests(requestReferences);
-                        /*
-                        Todo: Ask for public constructor
-                         */
+                if (category == null) {
+                    category = stringCateogryToURI(jsonAttribute.getKey());
+                }
+                if (jsonCategory.has(EntitlementEndpointConstants.ID)) {
+                    id = jsonCategory.get(EntitlementEndpointConstants.ID).getAsString();
+                }
+                if (jsonCategory.has(EntitlementEndpointConstants.CONTENT)) {
+                    DocumentBuilderFactory dbf;
+                    Document doc = null;
+
+                    String xmlContent = stringContentToXMLContent(jsonCategory
+                            .get(EntitlementEndpointConstants.CONTENT)
+                            .getAsString());
+                    dbf = IdentityUtil.getSecuredDocumentBuilderFactory();
+                    dbf.setNamespaceAware(true);
+
+                    try (
+                            ByteArrayInputStream inputStream = new ByteArrayInputStream(xmlContent.getBytes())
+                    ){
+                        doc = dbf.newDocumentBuilder().parse(inputStream);
+                    } catch (Exception e) {
+                        throw new JsonParseException("DOM of request element can not be created from String" + e.getMessage());
```

Comment:
pass 'e' as a parameter
end the message with '.'
e.getMessage() will be redundant when 'e' is passed, so consider removing it